### DISCUSSION TITLE
能否通过DSL完成Demo到组件的Glue?

### DISCUSSION THREAD

**Discussion by lowkeyrd** [Serverless-Devs/Serverless-Devs#323] [General]
## 背景
现在的很多框架类的demo，比如start-xxx(start-thinkphp/start-express)，干的活主要是：1. 内置框架；2. 适配的s.yaml，主要起到的是脚手架的工作。每个start-xxx，s.yaml中几乎都引用了一个对应的xxx组件，比如start-thinkphp，引用了thinkphp组件，但是这些被应用的组件基本没做什么独立的事情，都是加载web-framework组件或者fc组件执行相应的命令，这样就导致开发一个demo的灵活度不足，需要单独开发一个xxx组件来进行胶水层的工作

## 讨论
既然`start-xxx` -> `xxx` -> `web-framework` 的实现高度一致，能否通过DSL方式自动完成`xxx`干的事情，这样可以省掉一层组件的开发，让更多开发者贡献自己的demo

**Comment by lowkeyrd** [comment]
这个问题可以保持开放性讨论，有具体的设计建议会贴出来

**Comment by anycodes** [comment]
我觉得这个就是我们即将要做的web-framework的设计了。我们要讨论一起看一下的。
